### PR TITLE
Migrate Timestamps

### LINKED ISSUE #275: Move dates to timestamps
Move all stored dates to timestamps

### PR DESCRIPTION
**Pull request by jotoeri**
- Moving all dates to unix-timestamps.
- Renamed expirationDate now to expiresTimestamp
- Defined `expiresTimestamp==0` as 'form does not expire/`expires==false`'
- Created the procedures getExpires() & getExpired() for the computed properties of the form.
- Didn't take too much effort into the sidebar now due to the pending rewrite, just so it works for now.

closes #275

### CODE REVIEW

File: lib/Db/Form.php [hunk 1]
```diff
@@ -101,9 +105,9 @@ public function read() {
 			'ownerDisplayName' => \OC_User::getDisplayName($this->getOwnerId()),
 			'created' => $this->getCreated(),
 			'access' => $this->getAccess(),
-			'expires' => $expires,
-			'expired' => $expired,
-			'expirationDate' => $this->getExpirationDate(),
+			'expires' => $this->getExpires(),
+			'expired' => $this->getExpired(),
+			'expiresTimestamp' => $this->getExpiresTimestamp(),
```
Comment (skjnldsv) [inline lib/Db/Form.php:110]: What is the diff between expires & expiresTimestamp ?
`expires` should be enough, no?

Comment (jotoeri) [reply lib/Db/Form.php:110]: `expires` is the boolean value, that is modeled by the checkbox. But yes, maybe we could move that in a further cleanup to be watched & calculated just where it is necessary... Same for `expired`, that is also a calculated boolean.

Comment (skjnldsv) [reply lib/Db/Form.php:110]: I think that is enough to store the timestamp too yes :)

Comment (jotoeri) [reply lib/Db/Form.php:110]: Let's do this when cleaning up the form-object anyways....?

File: lib/Migration/Version010200Date2020323141300.php [hunk 2]
```diff
@@ -220,8 +225,8 @@ public function postSchemaChange(IOutput $output, \Closure $schemaClosure, array
 						'description' => $qb_restore->createNamedParameter($event['description'], IQueryBuilder::PARAM_STR),
 						'owner_id' => $qb_restore->createNamedParameter($event['owner'], IQueryBuilder::PARAM_STR),
 						'access_json' => $qb_restore->createNamedParameter($newAccessJSON, IQueryBuilder::PARAM_STR),
-						'created' => $qb_restore->createNamedParameter($event['created'], IQueryBuilder::PARAM_STR),
-						'expiration_date' => $qb_restore->createNamedParameter($event['expire'], IQueryBuilder::PARAM_STR),
+						'created' => $qb_restore->createNamedParameter($this->convertDateTime($event['created']), IQueryBuilder::PARAM_INT),
+						'expires_timestamp' => $qb_restore->createNamedParameter($this->convertDateTime($event['expire']), IQueryBuilder::PARAM_INT),
```
Comment (skjnldsv) [inline lib/Migration/Version010200Date2020323141300.php:229]: So I might be picky here, but I actually don't like using variable names that describe themselves :p
it's like doing `submit_once_boolean`, doesn't make sense to me :see_no_evil: I would go for just simple `expires`, same we do with `created` :grin:

Comment (jotoeri) [reply lib/Migration/Version010200Date2020323141300.php:229]: Yes, you're right, but this is the same as we just discussed above. Expires is used by the other expires variable. So when cleaning up the form-object, i will rename this one to `expires` too.

Comment (skjnldsv) [reply lib/Migration/Version010200Date2020323141300.php:229]: You'll clean the migration too?

Comment (jotoeri) [reply lib/Migration/Version010200Date2020323141300.php:229]: Well, when renaming the object-variable, DB should be changed consistently too, yes.

Comment (skjnldsv) [reply lib/Migration/Version010200Date2020323141300.php:229]: Good to me :)

**Pull request merged**